homeassistant/components/amberelectric/sensor.py

```diff
@@ -104,6 +104,18 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
             data["range_min"] = format_cents_to_dollars(interval.range.min)
             data["range_max"] = format_cents_to_dollars(interval.range.max)
 
+        if interval.advanced_price is not None:
+            multiplier = -1 if interval.channel_type == ChannelType.FEEDIN else 1
+            data["advanced_price_low"] = multiplier * format_cents_to_dollars(
```

Comment:
We don't allow using a state attribute to represent the measurement if the measurement is significant without knowing the main state of the entity. If a measurement is significant on its own it should be a separate (sensor) entity.

Also, we don't allow forecasts in the state machine. For those kind of measurements, a service with a service response should be used.

https://developers.home-assistant.io/docs/dev_101_services/#response-data

Reply:
I get that. This is an attribute of the price though, it's not the state (the state is the perKWH price).

Regarding the forecast, I haven't updated the forecast sensor to include this attribute, but removing the whole sensor will be a breaking change, and I need to build the service replacement before I can deprecate it.

Reply:
I don't understand what you mean with this:
> This is an attribute of the price though, it's not the state (the state is the perKWH price).

The problem is that you're adding it as a state attribute of a state instead of a separate state (a separate entity) (if it wasn't a forecast).

(It should not be part of a state at all if it's a forecast)

Reply:
I suggest you add a custom service that returns forecast data. Deprecation of existing state or state attributes can be done separately.

Reply:
This is the current price sensor, not the forecast sensor.

The forecast sensor is below this, and I haven't touched that in this PR for the reasons you've outlined.

Perhaps some background that may help. https://help.amber.com.au/hc/en-us/articles/360044364852-Understanding-30-minute-settlements-and-AEMO-prices

It's a good enough estimate for automation, which is why it's presented as the current price in our app and via our API. And we use it as the source of pricing for our internal automation platform.

So for all intents and purposes it _isn't_ a forecast, it _is_ the price.

The advanced price uses a different model for the forecasts, so gives a more accurate "current price", which is why I've presented it here as an attribute rather than a sensor.

After January, For customers that do support 5 minute settlement this attribute won't be shown, because the prices they will see will be actual 5 minute prices, but for those on 15 or 30 minute meters, they will still see the advanced prices attribute in the first 25 minutes.

Reply:
Reading this back, i see your point about it being a sensor. I will refactor.

Reply:
Ok, good.